plugins/providers/shield/provider.go

```diff
@@ -125,22 +125,35 @@ func (p *provider) addOrganizations(pc *domain.ProviderConfig, organizations []*
 	return resources
 }
 
-func (p *provider) getClient(providerURN string, credentials Credentials) (ShieldClient, error) {
+func (p *provider) GetClient(providerURN string, credentials Credentials) (ShieldClient, error) {
 	if p.Clients[providerURN] != nil {
 		return p.Clients[providerURN], nil
 	}
 
-	client, err := NewClient(&ClientConfig{
-		Host:       credentials.Host,
-		AuthHeader: credentials.AuthHeader,
-		AuthEmail:  credentials.AuthEmail,
-	}, p.logger)
-	if err != nil {
-		return nil, err
-	}
+	if credentials.ClientVersion == "new" {
```

Comment:
instead of "new" let's go with "v2"?

Reply:
But it's not any specific version i.e. `v2` or `v3` we are upgrading it to. 
It is a new shield for us and that new shield versions may also change with changing time.

Therefor renaming it to v2 looks like some exact version of shield, CMIIW?
and also once our shield deployment get's stable, we'll remove this logic and keep only one shield client.